test_conformance/commonfns/test_unary_fn.cpp

```diff
@@ -207,33 +225,38 @@ int test_unary_fn(cl_device_id device, cl_context context,
                 get_random_double(-100000.0 * M_PI, 100000.0 * M_PI, d);
         }
     }
+    else if (std::is_same<T, half>::value)
+    {
+        pragma_str = "#pragma OPENCL EXTENSION cl_khr_fp16 : enable\n";
+        for (int j = 0; j < num_elements; j++)
+        {
+            input_ptr[j] = conv_to_half(get_random_float(
+                (float)(-65503.f * M_PI), (float)(65503.f * M_PI), d));
```

Comment:
for the other tests we've just copied the random float range from the float and double tests (even when it's a hex values like `0x20000000` I don't really understand). Why `-65503.f` here?

Reply:
`10000.0 * M_PI` is more in line with the multiplier used in the float/double tests, so happy enough with that. Though I can't say I know why `+/- 100000.0 * M_PI` is used, I don't have any better suggestions for moving forward here without reverse engineering how all the random input limits are set.